src/Library/demos/Preferences Window/main.blp

```diff
@@ -0,0 +1,160 @@
+        title: "Use Dark Mode";
+        activatable-widget: dm_switch;
+
+        [suffix]
+        Switch dm_switch{
+          halign: center;
+          valign: center;
+        }
+      }
+    }
+
+    Adw.PreferencesGroup {
+      title: "Text Settings";
+      description: "Customize the appearance of text in the application.";
+      //
+
+      Adw.ActionRow {
+        title: "Font Size";
+
+        [suffix]
+        Gtk.SpinButton{
+          halign: center;
+          valign: center;
+          adjustment:
+          Gtk.Adjustment size_adj {
+            lower: 5;
+            upper: 999;
+            step-increment: 1;
+            value: 11;
+          };
+        }
+      }
+
+      Adw.ActionRow {
+        title: "Font Color";
+
+        [suffix]
+        Gtk.ColorDialogButton{
+          halign: center;
+          valign: center;
+          dialog: ColorDialog {};
+        }
+      }
+
+      LinkButton {
+        label: "API Reference";
+        uri: "https://gnome.pages.gitlab.gnome.org/libadwaita/doc/1.3/class.PreferencesWindow.html";
+      }
+    }
+  }
+
+  Adw.PreferencesPage behavior_page {
+    title:_("Behavior");
+    icon-name:"settings-symbolic";
+
+
+    Adw.PreferencesGroup{
+      title:_("Interaction Settings");
+      description:"Change how the app behaves during user interaction.";
+
+      //Adw.PreferencesGroup can have suffix widgets like Adw.ActionRows
+      [header-suffix]
+      Button{
+        halign: center;
+        valign: center;
+        icon-name: "settings-symbolic";
+      }
+
+      Adw.ActionRow{
+        title: "Run on Startup";
+        activatable-widget: startup_switch;
+
+        [suffix]
+        Switch startup_switch{
+          halign: center;
+          valign: center;
+        }
+      }
+      Adw.ActionRow{
+        title: "Show Toast";
+        activatable-widget: toast_button;
+
+        [suffix]
+        Button toast_button{
+          halign: center;
+          valign: center;
+          label: "show toast";
+          icon-name: "bread-symbolic";
+        }
+      }
+    }
+
+    Adw.PreferencesGroup{
+      title:"Data Settings";
+      description:"Manage user data related settings.";
+
+
+      Adw.ActionRow{
+        title: "Enable Telemetry";
+        activatable-widget: telemetry_switch;
+
+        [suffix]
+        Switch telemetry_switch{
+          halign: center;
+          valign: center;
+        }
+      }
+
+      Adw.ActionRow{
+        title: "Enable Auto-Updates";
+        activatable-widget: update_switch;
+
+        [suffix]
+        Switch update_switch{
+          halign: center;
+          valign: center;
+        }
+      }
+
+      Adw.ActionRow subpage_row{
+        title: "Additional Preferences";
+        activatable: true;
+
+        [suffix]
+        Gtk.Image{
+          icon-name: "arrow1-right-symbolic";
+        }
+      }
+    }
+  }
+}
+
+Adw.StatusPage subpage {
+  title: "Preferences windows can have subpages like this one";
```

Comment:
I'd suggest changing this to "Custom Subpage" or something like that, and move the existing label to the `description` property.